### PR TITLE
Yvr client 39 - Don't merge

### PR DESCRIPTION
**Pull request by gdcho**
Google authentication for iOS, Android, and web

Steps to replicate:

1. Install the following dependencies

```
npx expo install expo-auth-session expo-crypto expo-web-browser react-native-web react-dom @expo/webpack-config @react-native-async-storage/async-storage
```

2. Create API Keys for Android, iOS, and web on[ Google Cloud Platform](https://console.cloud.google.com/apis/credentials)

- Website:
 Authorized JavaScript origins and redirect URIs `https://localhost:8080`

- iOS and Android:
`npx expo prebuild` to generate bundleIdentifier ID (iOS) and package name (Android)

- Android (Generate SHA-1 certificate fingerprint)
Run `npx expo credentials:manager`, Select platform android, Update upload keystore, Generate new keystore, copy `Google Certificate Hash (SHA-1)` 

3. Copy API keys and place in .env and run `npm i react-native-dotenv` to install dotenv and import like:

```
import { ANDROID_CLIENT_ID, IOS_CLIENT_ID, WEB_CLIENT_ID } from "@env";
```

4. Refer to branch code for implementation of Google SignIn

5. Test by running `npx expo run` `web` `android` or `ios`

### CODE REVIEW

File: app/(tabs)/index.tsx [hunk 1]
```diff
@@ -1,13 +1,75 @@
...
+import EditScreenInfo from "@/components/EditScreenInfo";
+import { Text, View } from "@/components/Themed";
+import * as WebBrowser from "expo-web-browser";
+import * as Google from "expo-auth-session/providers/google";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { ANDROID_CLIENT_ID, IOS_CLIENT_ID, WEB_CLIENT_ID } from "@env";
 
-import EditScreenInfo from '@/components/EditScreenInfo';
-import { Text, View } from '@/components/Themed';
+WebBrowser.maybeCompleteAuthSession();
```
Comment (jasper-oh) [inline app/(tabs)/index.tsx:10]: Can you give me brief info about this line

File: app/(tabs)/index.tsx [hunk 2]
```diff
@@ -1,13 +1,75 @@
...
+
+  React.useEffect(() => {
+    handleGoogleSignIn();
+  }, [response]);
+
+  async function handleGoogleSignIn() {
+    const user = await AsyncStorage.getItem("@user");
+    if (!user) {
+      if (response?.type === "success") {
+        console.log("success!!");
```
Comment (jasper-oh) [inline app/(tabs)/index.tsx:28]: Remove console when you merge thanks

**Review by jasper-oh — APPROVED** [review]
Thanks for your hard work

File: app/(tabs)/index.tsx (hunk 1 above)
Comment (gdcho) [reply app/(tabs)/index.tsx:10]: Used for handling redirects: https://docs.expo.dev/versions/latest/sdk/webbrowser/

**Comment by gdcho** [comment]
[[YVR-Client-39] Google OAuth Authentication](https://trello.com/c/WIkQ31rA/39-yvr-client-39-google-oauth-authentication)

**Pull request closed without merging**